test/e2e/e2e_test.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// kubeconfigPath is path to the kubernetes configuration file.
+var kubeconfigPath = filepath.Join(os.Getenv("HOME"), "/.kube/config")
+
+// newKubeClient create new Kubernetes client instance based on kubeconfig file.
+func newKubeClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
+	// use the current context in kubeconfig.
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the clientset.
+	return kubernetes.NewForConfig(config)
+}
+
+func newEtcdProxyClinet(kubeconfigPath string) (*clientset.Clientset, error) {
+	// use the current context in kubeconfig.
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the clientset.
+	return clientset.NewForConfig(config)
+}
+
+func replicasetName(etcdstorageName string) string {
+	return fmt.Sprintf("etcd-rs-%s", etcdstorageName)
+}
+
+func serviceName(etcdstorageName string) string {
+	return fmt.Sprintf("etcd-%s", etcdstorageName)
+}
+
+func TestDeployEtcdStorage(t *testing.T) {
+	client, err := newKubeClient(kubeconfigPath)
+	if err != nil {
+		t.Fatalf("unable to create kubernetes client from provided kubeconfig: %v", err)
+	}
+
+	etcdproxyClient, err := newEtcdProxyClinet(kubeconfigPath)
+	if err != nil {
+		t.Fatalf("unable to create etcdproxy client from provided kubeconfig: %v", err)
+	}
+
+	tests := []struct {
+		name                   string
+		etcdStorage            *v1alpha1.EtcdStorage
+		etcdStorageValid       bool
+		expectedReplicaSetName string
+		expectedReplicas       int32
+		expectedServiceName    string
+	}{
+		{
+			name: "test simple etcdstorage creation",
+			etcdStorage: &v1alpha1.EtcdStorage{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "es-test-1",
+				},
+			},
+			expectedReplicaSetName: replicasetName("es-test-1"),
+			expectedReplicas:       int32(1),
+			expectedServiceName:    serviceName("es-test-1"),
+			etcdStorageValid:       true,
+		},
+		{
+			name: "test etcdstorage creation - name too long",
+			etcdStorage: &v1alpha1.EtcdStorage{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "es-test-1-this-name-is-too-long-limit-foo-bar-baz-foo-bar-baz-123",
+				},
+			},
+			etcdStorageValid: false,
+		},
+	}
+	for _, tc := range tests {
+		es, err := etcdproxyClient.EtcdV1alpha1().EtcdStorages().Create(tc.etcdStorage)
+		if err != nil && tc.etcdStorageValid {
+			t.Fatalf("unable to create etcdstorage '%s': %v", tc.etcdStorage.Name, err)
+		}
+
+		// Only continue if etcdStorage is expected to be valid.
+		if tc.etcdStorageValid {
```

Comment:
I have a problem here.

We have defined that EtcdStorages name length should be less than 59 characters (64 which is max for Services, -5 for `etcd-` prefix). I added a test case for that, but, this is going to fail on `Create`. 

But that failure is expected, so I was not sure how to stop there. I added `etcdStorageValid` bool, and if it is false, error on Create is going to be ignored, and we're not going to continue.

Is there a better way to mark test as passed, than this one? There is 'FailNow()`, which fails, but I don't want fail`.
There is also `SkipNow()`, but it marks test as skipped, and I was not sure do we want that.